pkg/localenvoy/controllers/ingress/envoyingress.go

```diff
@@ -29,14 +29,14 @@ import (
 	"k8s.io/klog/v2"
 
 	envoycontrolplane "github.com/kcp-dev/kcp/pkg/localenvoy/controlplane"
-	nscontroller "github.com/kcp-dev/kcp/pkg/reconciler/workload/namespace"
+	"github.com/kcp-dev/kcp/pkg/syncer"
 )
 
 // reconcile is triggered on every change to an ingress resource.
 func (c *Controller) reconcile(ctx context.Context, ingress *networkingv1.Ingress) error {
 	klog.InfoS("reconciling Ingress", "ClusterName", logicalcluster.From(ingress), "Namespace", ingress.Namespace, "Name", ingress.Name)
 
-	if ingress.Labels[nscontroller.ClusterLabel] == "" {
+	if syncer.GetAssignedWorkloadCluster(ingress.Labels) == "" {
```

Comment:
wrong package. This is something from the workload API and should live in pkg/apis/workload/helpers or so.